### PR TITLE
Recipe to build libqt4pas under Haiku. This is a required dependency …

### PR DESCRIPTION
**Pull request by oco27**
…to build

Lazarus under Haiku.

### CODE REVIEW

**Comment by mmuman** [comment]
Looks like the CI detected trailing whitespace, like lines 24, 37, 45...
Also, commit logs should have a short first line, the second sentence should go into its own paragraph after an empty line.

File: dev-qt/libqt4pas/libqt4pas-2.5.0.recipe [hunk 1]
```diff
@@ -0,0 +1,71 @@
+SUMMARY="Pascal Qt interface"
+DESCRIPTION="Pascal wrapper around Qt interfaces.
+Just the necessary stuff for Lazarus is wrapped here."
+HOMEPAGE="http://users.telenet.be/Jan.Van.hijfte/qtforfpc/fpcqt4.html"
+SOURCE_URI="http://users.telenet.be/Jan.Van.hijfte/qtforfpc/V2.5/qt4pas-V2.5_Qt4.5.3.tar.gz"
+CHECKSUM_SHA256="825423db80da4df5c21816c0392b3394cddfe2f3293dfd08ace84941726affea"
+SOURCE_DIR="qt4pas-V2.5_Qt4.5.3"
+LICENSE="GNU LGPL v2.1"
```
Comment (fbrosson) [inline dev-qt/libqt4pas/libqt4pas-2.5.0.recipe]: The correct [order](https://github.com/haikuports/haikuports/wiki/HaikuPorter-Guidelines#ordering), after DESCRIPTION, is:
```
HOMEPAGE="http://users.telenet.be/Jan.Van.hijfte/qtforfpc/fpcqt4.html"
COPYRIGHT="2005-2011 Jan Van hijfte"
LICENSE="GNU LGPL v2.1"
REVISION="1"
SOURCE_URI="http://users.telenet.be/Jan.Van.hijfte/qtforfpc/V2.5/qt4pas-V2.5_Qt4.5.3.tar.gz"
CHECKSUM_SHA256="825423db80da4df5c21816c0392b3394cddfe2f3293dfd08ace84941726affea"
SOURCE_DIR="qt4pas-V2.5_Qt4.5.3"

ARCHITECTURES="!x86_gcc2 x86 x86_64"
SECONDARY_ARCHITECTURES="x86"
```

File: dev-qt/libqt4pas/libqt4pas-2.5.0.recipe [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+COPYRIGHT="2005-2011 Jan Van hijfte"
+REVISION="1"
+ARCHITECTURES="!x86_gcc2 x86 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+	libqt4pas$secondaryArchSuffix = $portVersion
+	lib:libqt4pas$secondaryArchSuffix = 5.2.5 compat >= 5.2
+	"
+
```
Comment (fbrosson) [inline dev-qt/libqt4pas/libqt4pas-2.5.0.recipe]: No need for an empty line here, between PROVIDES and REQUIRES.
BTW, the right order is:
```
PROVIDES
REQUIRES

PROVIDES_devel
REQUIRES_devel

BUILD_REQUIRES
BUILD_PREREQUIRES
```

File: dev-qt/libqt4pas/libqt4pas-2.5.0.recipe [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+BUILD_REQUIRES="
+	cmd:make
+	cmd:qmake$secondaryArchSuffix
+	cmd:g++$secondaryArchSuffix
+	lib:libqtcore$secondaryArchSuffix
+	devel:libqtcore$secondaryArchSuffix
+	"
+
+BUILD_PREREQUIRES="
+	haiku${secondaryArchSuffix}_devel
```
Comment (fbrosson) [inline dev-qt/libqt4pas/libqt4pas-2.5.0.recipe]: **`haiku${secondaryArchSuffix}_devel`** should go to the begining of BUILD_REQUIRES.
And **`lib:libqtcore$secondaryArchSuffix`** and **`devel:libqtcore$secondaryArchSuffix`** should also go to BUILD_REQUIRES.

File: dev-qt/libqt4pas/libqt4pas-2.5.0.recipe [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+REQUIRES="
+	haiku${secondaryArchSuffix}
+	lib:libqtcore$secondaryArchSuffix
+	"
+
+BUILD_REQUIRES="
+	cmd:make
+	cmd:qmake$secondaryArchSuffix
+	cmd:g++$secondaryArchSuffix
```
Comment (fbrosson) [inline dev-qt/libqt4pas/libqt4pas-2.5.0.recipe]: **`cmd:make`** and **`cmd:qmake$secondaryArchSuffix`** should only be in BUILD_PREREQUIRES.
**`cmd:g++$secondaryArchSuffix`** should go to BUILD_PREREQUIRES.

BTW, if you saw **`cmd:`** lines in the BUILD_REQUIRES of existing recipes then we need to fix them ;-)
Similarly, if you saw **`lib:`** or **`devel:`** lines in the BUILD_PREREQUIRES of existing recipes then we also need to fix them ;-)

File: dev-qt/libqt4pas/libqt4pas-2.5.0.recipe [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+	lib:libqtcore$secondaryArchSuffix
+	devel:libqtcore$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	cmd:gcc${secondaryArchSuffix}
+	cmd:ld${secondaryArchSuffix}
+	lib:libqtcore$secondaryArchSuffix
+	devel:libqtcore$secondaryArchSuffix
```
Comment (fbrosson) [inline dev-qt/libqt4pas/libqt4pas-2.5.0.recipe]: Thanks for your latest fixes. You're almost done ;-)
You forgot, however, a couple of things:

* **`lib:libqtcore$secondaryArchSuffix`** and **`devel:libqtcore$secondaryArchSuffix`** should be removed from BUILD_PREREQUIRES because we don't need them here.

* As explained in the comment above, the following 3 **`cmd:`** lines, currently in BUILD_REQUIRES, should be moved to BUILD_PREREQUIRES:
```
	cmd:make
	cmd:qmake$secondaryArchSuffix
	cmd:g++$secondaryArchSuffix
```
BTW, if you could sort alphabetically the **`cmd:`** lines in BUILD_PREREQUIRES it would be perfect!

* While at it, could you please also drop, in the whole recipe, a few curly braces where this is possible?
To do so, each occurence of «**`${secondaryArchSuffix}`**», when **_not_** followed by «**`_devel`**», should be replaced by «**`$secondaryArchSuffix`**».
In other words, we shouldn't have any «**`${secondaryArchSuffix}`**» but only «**`$secondaryArchSuffix`**» and «**`${secondaryArchSuffix}_devel`**».

File: dev-qt/libqt4pas/libqt4pas-2.5.0.recipe [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+	make 
+}
+
+INSTALL()
+{
+	make install
+	mkdir -p $libDir
+	cp libQt4Pas.so* $libDir
+	# prepare develop/lib
+	prepareInstalledDevelLibs libQt4Pas
```
Comment (fbrosson) [inline dev-qt/libqt4pas/libqt4pas-2.5.0.recipe]: Since there is only one lib, you can use prepareInstalledDevel**Lib** instead of  prepareInstalledDevel**Libs**:
```
	prepareInstalledDevelLib libQt4Pas
```

File: dev-qt/libqt4pas/libqt4pas-2.5.0.recipe [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+	cmd:qmake$secondaryArchSuffix
+	cmd:qmake
+	cmd:make
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	cmd:make
+	cmd:qmake$secondaryArchSuffix
+	cmd:g++$secondaryArchSuffix
```
Comment (fbrosson) [inline dev-qt/libqt4pas/libqt4pas-2.5.0.recipe]: Thanks for your latest fixes. You're almost done ;-)
You forgot, however, a couple of things:

* The following 3 **`cmd:`** lines, currently in BUILD_REQUIRES, should be moved to BUILD_PREREQUIRES:
```
	cmd:make
	cmd:qmake$secondaryArchSuffix
	cmd:g++$secondaryArchSuffix
```
